reladomo/src/main/java/com/gs/fw/common/mithra/attribute/calculator/arithmeticCalculator/DateDayOfMonthCalculator.java

```diff
@@ -71,4 +71,19 @@ public Operation optimizedIntegerEq(int value, CalculatedIntegerAttribute intAtt
     {
         return new IntegerEqOperation(intAttribute, value);
     }
+
+    public boolean equals(Object obj)
+    {
+        if (this == obj) return true;
+        if (obj.getClass().equals(this.getClass()))
+        {
+            return this.attribute.equals(((DateDayOfMonthCalculator)obj).attribute);
+        }
+        return false;
+    }
+
+    public int hashCode()
+    {
+        return 0x12345678 ^ this.attribute.hashCode();

```

Comment:
Please use different constants for each class.

Reply:
Sure, updated the constants by just rotating the digits. How do you pick them? 0x12345678 already appears in two places.

https://github.com/goldmansachs/reladomo/search?q=0x12345678&unscoped_q=0x12345678

Reply:
Ideally, just a random number. The idea is that they should be different for things that can possibly clash. The two cases of 0x12345678 (luckily?) can't : one is absolute value, the other is toLowerCase.